cow_instrument/MoveCommand.cpp

```diff
@@ -3,8 +3,10 @@
 
 MoveCommand::MoveCommand(V3D offset) : m_offset(offset) {}
 
-void MoveCommand::execute(Component &component) const {
-  component.deltaPos(m_offset);
+bool MoveCommand::execute(CowPtr<Component> &component) const {
+  component->deltaPos(m_offset); // De-referenceing forces a copy
+  return true; // If you're moving, components are being written to. Copy
+               // happened on previous line.
```

Comment:
Dereferences only forces copy if the reference count is larger than 1. You want to return true only if there really was a copy, otherwise you can keep the detector vector and avoid that overhead.